app/workers/twitter_checkin_share_worker.rb

```diff
@@ -2,106 +2,75 @@ class TwitterCheckinShareWorker
     end
 
-    twitter = @checkin.user.providers.where(name: 'twitter').first
+    twitter = @checkin.user.providers.find_by(name: "twitter")
     client = Twitter::REST::Client.new do |config|
-      config.consumer_key        = ENV['TWITTER_CONSUMER_KEY']
-      config.consumer_secret     = ENV['TWITTER_CONSUMER_SECRET']
-      config.access_token        = twitter.token
+      config.consumer_key = ENV["TWITTER_CONSUMER_KEY"]
+      config.consumer_secret = ENV["TWITTER_CONSUMER_SECRET"]
+      config.access_token = twitter.token
       config.access_token_secret = twitter.token_secret
     end
 
-    client.update(tweet_text(@checkin.comment))
+    client.update(tweet_body(@checkin))
   end
 
 
   private
 
-  def get_work_title
-    max_length = 35
-    title = @checkin.work.title
-
-    if @checkin.comment.present?
-      if max_length < title.length
-        title.truncate(max_length)
+  def tweet_body(checkin)
+    data = {
+      work_title: checkin.work.title,
+      episode_number: checkin.episode.single? ? "" : checkin.episode.number,
+      episode_title: get_episode_title(checkin),
+      share_url: "#{ENV['ANNICT_URL']}/r/tw/#{checkin.twitter_url_hash}",
+      share_hashtag: checkin.work.twitter_hashtag.presence || "",
+      comment: checkin.comment
+    }
+
+    body = generate_tweet_body(data)
+
+    if body.length > 140
+      if data[:comment].present?
+        data[:comment] = truncate_text(data[:comment], body)
       else
-        @rest += (max_length - title.length)
-        title
+        data[:episode_title] = truncate_text(data[:episode_title], body)
+        body = generate_tweet_body(data)
+
+        if body.length > 140
+          data[:work_title] = truncate_text(data[:work_title], body)
+        end
       end
-    else
-      title.truncate(max_length)
     end
-  end
-
-  def get_episode_number
-    max_length = 10
-    number = @checkin.episode.single? ? '' : @checkin.episode.number
 
-    if max_length < number.length
-      number.truncate(max_length)
-    else
-      @rest += (max_length - number.length)
-      number
-    end
+    generate_tweet_body(data)
   end
 
-  def get_episode_title
-    title = if @checkin.episode.single?
-      ''
-    else
-      @checkin.episode.title.presence || ''
-    end
-
-    if @checkin.comment.present?
-      @rest += 10
+  def get_episode_title(checkin)
+    if checkin.episode.single?
+      ""
     else
-      max_length = 30
-      title = if max_length < title.length
-        title.truncate(max_length)
-      else
-        title
-      end
-
-      title.present? ? "「#{title}」" : ' '
+      checkin.episode.title.presence || ""
     end
   end
 
-  def get_share_url
-    "#{ENV['ANNICT_URL']}/r/tw/#{@checkin.twitter_url_hash}"
-  end
-
-  def get_share_hashtag
-    max_length = 20
-    hashtag = @checkin.work.twitter_hashtag.presence || ''
-
-    if max_length < hashtag.length
-      hashtag.truncate(max_length)
+  def generate_tweet_body(data)
```

Comment:
Assignment Branch Condition size for generate_tweet_body is too high. [16.16/15]